libs/lwt_platform/Lwt_platform.mli

```diff
@@ -13,3 +13,9 @@ val set_engine : unit -> unit
     This is important as select/poll is not great, and can easily
     run out of FDs, crashing the LS.
   *)
+
+val timeout : float -> 'a Lwt.t
+(** [timeout delay] returns a promise that will raise an exception after [delay] seconds. *)
```

Comment:
Can we document which exception? We may need to declare an exception here and then map whatever is raised by, e.g., `Lwt_unix` to `Lwt_platform.Timeout`.